controllers/knative/knative.go

```diff
@@ -84,3 +116,95 @@ func GetKnativeAvailability(cfg *rest.Config) (*Availability, error) {
 		return result, nil
 	}
 }
+
+func GetWorkflowSink(workflow *operatorapi.SonataFlow, pl *operatorapi.SonataFlowPlatform) (*duckv1.Destination, error) {
+	if workflow == nil {
+		return nil, nil
+	}
+	if workflow.Spec.Sink != nil {
+		return workflow.Spec.Sink, nil
+	}
+	if pl != nil && pl.Spec.Eventing != nil {
+		// no sink defined in the workflow, use the platform broker
+		return pl.Spec.Eventing.Broker, nil
+	} else if pl.Status.ClusterPlatformRef != nil {
+		// Find the platform referred by the cluster platform
+		platform := &operatorapi.SonataFlowPlatform{}
+		if err := utils.GetClient().Get(context.TODO(), types.NamespacedName{Namespace: pl.Status.ClusterPlatformRef.PlatformRef.Namespace, Name: pl.Status.ClusterPlatformRef.PlatformRef.Name}, platform); err != nil {
+			return nil, fmt.Errorf("error reading the platform referred by the cluster platform")
+		}
+		if platform.Spec.Eventing != nil {
+			return platform.Spec.Eventing.Broker, nil
+		}
+	}
+	return nil, nil
+}
+
+func IsKnativeBroker(kRef *duckv1.KReference) bool {
+	return kRef.APIVersion == "eventing.knative.dev/v1" && kRef.Kind == "Broker"
```

Comment:
can you please move these to private `const`s?

Reply:
Done